app/controllers/IsThisYourBusinessController.scala

```diff
@@ -60,35 +58,22 @@ class IsThisYourBusinessController @Inject() (
     extends FrontendBaseController
     with I18nSupport
     with NunjucksSupport
-    with WithEitherT
     with Logging {
 
   private val form = formProvider()
 
-  private def result(mode: Mode, regime: Regime, form: Form[Boolean])(implicit ec: ExecutionContext, request: DataRequest[AnyContent]) =
-    (for {
-      registrationInfo <- EitherT(matchBusinessInfo(regime))
-      updatedAnswers   <- setEither(RegistrationInfoPage, registrationInfo)
-      _ = sessionRepository.set(updatedAnswers)
-    } yield registrationInfo)
-      .fold(
-        fa = {
-          case NotFoundError =>
-            Future.successful(Redirect(routes.NoRecordsMatchedController.onPageLoad(regime)))
-          case error =>
-            renderer.renderError(error, regime)
-        },
-        fb =>
-          subscriptionService.getDisplaySubscriptionId(regime, fb.safeId) flatMap {
-            case Some(subscriptionId) => controllerHelper.updateSubscriptionIdAndCreateEnrolment(fb.safeId, subscriptionId, regime)
-            case _ =>
-              val name     = fb.name.getOrElse("")
-              val address  = fb.address.getOrElse(AddressResponse("", None, None, None, None, ""))
-              val withForm = request.userAnswers.get(IsThisYourBusinessPage).fold(form)(form.fill)
-              render(mode, regime, withForm, name, address).map(Ok(_))
-          }
-      )
-      .flatten
+  private def result(mode: Mode, regime: Regime, form: Form[Boolean], registrationInfo: RegistrationInfo)(implicit
+    ec: ExecutionContext,
+    request: DataRequest[AnyContent]
+  ) =
+    subscriptionService.getDisplaySubscriptionId(regime, registrationInfo.safeId) flatMap {
+      case Some(subscriptionId) => controllerHelper.updateSubscriptionIdAndCreateEnrolment(registrationInfo.safeId, subscriptionId, regime)
+      case _ =>
+        val name     = registrationInfo.name.getOrElse("")
+        val address  = registrationInfo.address.getOrElse(AddressResponse("", None, None, None, None, ""))
+        val withForm = request.userAnswers.get(IsThisYourBusinessPage).fold(form)(form.fill)
+        render(mode, regime, withForm, name, address).map(Ok(_))
+    }
```

Comment:
Is this was discussed to address or was Ashwini Addressing this Issue?
AddressResponse("", None, None, None, None, "") This is all  not necessary

Reply:
Ashwini asked me to leave this (and other related changes) to DAC6-1297.